src/RichText/RichText.php

```diff
@@ -339,12 +339,26 @@ private static function replaceImagesByGallery(string $content): string
             if ($document->getFromDB($docid)) {
                 $docpath = GLPI_DOC_DIR . '/' . $document->fields['filepath'];
                 if (Document::isImage($docpath)) {
+                    //find width / height define by user
+                    $width = null;
+                    if (preg_match("/width=[\"|'](\d+)(\.\d+)?[\"|']/", $img_tag, $wmatches)) {
+                        $width = intval($wmatches[1]);
+                    }
+                    $height = null;
+                    if (preg_match("/height=[\"|'](\d+)(\.\d+)?[\"|']/", $img_tag, $hmatches)) {
+                        $height = intval($hmatches[1]);
+                    }
+
+                    //find real size from image
                     $imgsize = getimagesize($docpath);
+
                     $gallery = self::imageGallery([
                         [
                             'src' => $docsrc,
-                            'w'   => $imgsize[0],
-                            'h'   => $imgsize[1]
+                            'w'   => $imgsize[0], //used by photoSwipe gallery
+                            'h'   => $imgsize[1], //used by photoSwipe gallery
+                            'user_w' => $width, //use from timeline
+                            'user_h' => $height,//use from timeline
```

Comment:
`thumbnail` prefix will be easier to understand. No need to comment it.
```suggestion
                            'w'   => $imgsize[0],
                            'h'   => $imgsize[1],
                            'thumbnail_w' => $width,
                            'thumbnail_h' => $height,
```